src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/UrlRewriter.java

```diff
@@ -232,8 +273,61 @@ private static URL prefixWithProtocol(String url, String protocol) {
     }
   }
 
+  /**
+   * Create a new {@link Credentials} object by parsing the .netrc file with following order to
+   * search it:
+   *
+   * <ol>
+   *   <li>If environment variable $NETRC exists, use it as the path to the .netrc file
+   *   <li>Fallback to $HOME/.netrc
+   * </ol>
+   *
+   * @return the {@link Credentials} object or {@code null} if there is no .netrc file.
+   * @throws IOException in case the credentials can't be constructed.
+   */
+  //TODO : consider re-using RemoteModule.newCredentialsFromNetrc
+  @VisibleForTesting
+  static Credentials newCredentialsFromNetrc(Map<String, String> clientEnv, FileSystem fileSystem)
+          throws UrlRewriterParseException {
+    String netrcFileString =
+            java.util.Optional.ofNullable(clientEnv.get("NETRC"))
+                    .orElseGet(
+                            () ->
+                                    java.util.Optional.ofNullable(clientEnv.get("HOME"))
```

Comment:
On Windows, it should be `USERPROFILE `.
See https://cs.opensource.google/bazel/bazel/+/master:tools/build_defs/repo/http.bzl;l=88?q=%22USERPROFILE%22&ss=bazel